magento/pwa-studio
# [PWA-1784] [bug]  - Fix SearchBar blocking click events on filter/sort buttons

## Description

The SearchBar component has a `mousedown` event listener on the document to hide itself when clicking outside the area of its container element. This was blocking click events on `FilterItem` buttons and `ProductSort` because the page height would change after hiding searchBar and pointer location would invalidate `mouseup` events and not fire `onclick`.

This PR fixes these issues by using using `mousedown` event handlers on both Filter and Sorting buttons so it will fire always regardless of the page movement. 

## Related Issue







Closes PWA-1784

## Acceptance





### Verification Stakeholders



### Specification




## Verification Steps






#### Test scenario(s) for direct fix/feature





1. Search for `Tops`.
2. Click on the Price label for 100-200.
3. Ensure that filter is selected.
4. Search  for `Dress`.
5. Click on sorting dropdown.
6. Select a sorting option.
7. Ensure that product list is sorted accordingly. 

#### Test scenario(s) for any existing impacted features/areas

Ensure modified buttons are accessible through keyboard (Tab + Enter/Space) 




#### Test scenario(s) for any Magento Backend Supported Configurations





#### Is Browser/Device testing needed?





#### Any ad-hoc/edge case scenarios that need to be considered?




## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)





## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
